apps/src/studio/studio.js

```diff
@@ -858,10 +869,24 @@ Studio.onTick = function() {
     performItemOrProjectileMoves(Studio.items);
   }
 
+  Studio.updateFloatingScore();
+
   sortDrawOrder();
 
-  if (!spritesNeedMoreAnimationFrames && checkFinished()) {
-    Studio.onPuzzleComplete();
+  if (!Studio.succeeded && checkFinished()) {
+    Studio.succeeded = true;
```

Comment:
Could you just have one variable called `succeededTime`? If it is undefined, you haven't succeeded. If you have a time, then you have both the state (succeeded) and the time in one place.